src/test/test_ep_cred_client.c

```diff
@@ -140,16 +152,42 @@ test_run()
 		assert(rc == 0);
 
 		input = crt_req_get(rpc);
-		if (i == 0)
-			input->pi_delay = 1;
-		else
+
+		if (i == 0) {
+			/* If we test 'send to front of queue' flag we
+			 * want to increase response delay of first rpc to 3
+			 * seconds in order to allow sufficient queue up
+			 */
+			if (test.tg_send_queue_front)
+				input->pi_delay = 3;
+			else
+				input->pi_delay = 1;
+		} else {
 			input->pi_delay = 0;
+		}
 
 		rc = crt_req_send(rpc, rpc_handle_reply, NULL);
 		assert(rc == 0);
 		sent_count++;
 	}
 
+	/* Send RPC message to be in front of the queue. This option
+	 * should only be used when test.tg_burst_count is large while
+	 * test.tg_credits is small, allowing sufficient queue up of
+	 * rpcs
+	 */
+	if (test.tg_send_queue_front) {
+		rc = crt_req_create(test.tg_crt_ctx, &ep, OPC_PING_FRONT,
+				&rpc);
+		assert(rc == 0);
+
+		rc = crt_req_send(rpc, rpc_handle_ping_front_q, NULL);
+		assert(rc == 0);
+
+		test_sem_timedwait(&test.tg_front_queue_token, 61, __LINE__);
+		assert(sent_count != resp_count);
```

Comment:
Same comment about assert message for these three asserts.

Reply:
yep, will fix